modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+<?php
+class Jetpack_Widget_Social_Icons extends WP_Widget {
+	/**
+	 * @var array Default widget options.
+	 */
+	protected $defaults;
+
+	/**
+	 * @var array protocols that are allowed in esc_url validation function.
+	 */
+	protected $protocols = array( 'http', 'https', 'mailto', 'skype' );
+
+	/**
+	 * Widget constructor.
+	 */
+	public function __construct() {
+		$widget_ops = array(
+			'classname'                   => 'jetpack_widget_social_icons',
+			'description'                 => esc_html__( 'Add social-media icons to your site.', 'jetpack' ),
+			'customize_selective_refresh' => true,
+		);
+
+		parent::__construct( 'jetpack_widget_social_icons', esc_html__( 'Social Icons', 'jetpack' ), $widget_ops );
+
+		$this->defaults = array(
+			'title'     => esc_html__( 'Follow Us', 'jetpack' ),
+			'icon-size' => 'medium',
+			'icons'     => array(
+				array(
+					'url'   => '',
+				),
+			),
+		);
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_scripts' ) );
+		add_action( 'admin_print_footer_scripts', array( $this, 'render_admin_js' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_icon_scripts' ) );
+		add_action( 'wp_footer', array( $this, 'jetpack_social_menu_include_svg_icons' ), 9999 );
+	}
+
+	/**
+	 * Script & styles for admin widget form.
+	 */
+	public function enqueue_admin_scripts() {
+		wp_enqueue_script( 'jetpack-widget-social-icons-script', plugins_url( 'social-icons/social-icons-admin.js', __FILE__ ), array( 'jquery-ui-sortable' ), '20170506' );
+		wp_enqueue_style( 'jetpack-widget-social-icons-admin', plugins_url( 'social-icons/social-icons-admin.css', __FILE__ ), array(), '20170506' );
+	}
+
+	/**
+	 * Styles for front-end widget.
+	 */
+	public function enqueue_icon_scripts() {
+		wp_enqueue_style( 'jetpack-widget-social-icons-styles', plugins_url( 'social-icons/social-icons.css', __FILE__ ), array(), '20170506' );
```

Comment:
Again, overbroad enqueueing it looks like?  Let's try just registering things, and then actually enqueueing them when the code that uses them is run.